### DISCUSSION TITLE
Event fire before any requests?

### DISCUSSION THREAD

**Discussion by saulsharma** [unpoly/unpoly#415] [Q&A]
Hello, 
I'm trying to migrate away from Turbolinks to Unpoly. So far it's been good experience except for one of the issue. 

Turbolinks has a eventListener called `turbolinks:request-start` that is fired before making any requests. This allows us to inject custom headers before requesting next request. We currently use this when a webapp is loaded from mobile app. The code looks something like this:

```document.addEventListener("turbolinks:request-start", function(event) {
  var xhr = event.data.xhr;
  //check if xhr has header with the name is
  xhr.setRequestHeader("isNativeApp", '{{ request()->header('isNativeApp') }}');
  xhr.setRequestHeader("native_platform", '{{ request()->header('native_platform') }}');
  xhr.setRequestHeader("loadedFromNativeAppModal", '{{ request()->header('loadedFromNativeAppModal') }}');

});
```

Now, I'm trying to find a similar event for unpoly but so far, its been unsuccessful. I've tried with up:proxy:loaded, up:proxy:load etc but none of them seem to fire as expected. 
What would be the alternative of `turbolinks:request-start` for unpoly? 
Thanks!

**Comment by iniznet** [comment]
What unpoly version do you use? v1 does have that events as experimental, but in v2 it's renamed to [`up:request:load`](https://unpoly.com/up:request:load)

**Reply by saulsharma** [reply]
This seem to be working well! Thank you. 
But one issue I found here was, when some dynamic HTML content were created(for example, loaded a popup modal), the link doesn't trigger `up:request:load`.
I've already set `up.link.config.followSelectors.push('a[href]');` so every link should be loaded by unpoly by default, no?
I also tried adding up-follow on any link on modal, they just don't follow the link when a dynamic HTML is attached to body at all. 
Any solution in this case?

**Reply by adam12** [reply]
If it's not Unpoly doing the dynamic HTML, you'll need to notify Unpoly to look at the element with [`up.hello(el)`](https://unpoly.com/up.hello).

**Reply by saulsharma** [reply]
@adam12 I ended up using a different library for a modal, so the modal is fine now I reckon.
The `up.hello(el)` doesn't seem to update <meta content="theme-color"> though. I've also tried via 
`<meta name="theme-color" up-hungry content="#SomeColorCode">` but its always 1 step late. For example, our webapp has almost every pages with different theme-color but every new pages on navigation seem to load the previous one and the next one would replace the current meta color. What could be the reason in this case?
